python-frontend/src/main/java/org/sonar/python/semantic/v2/converter/PythonTypeToDescriptorConverter.java

```diff
@@ -120,9 +120,8 @@ private static Descriptor convert(String moduleFqn, String parentFqn, String sym
     var hasSuperClassWithoutDescriptor = false;
     var superClasses = new ArrayList<String>();
     for (var superClassWrapper : type.superClasses()) {
-      var superClass = superClassWrapper.type();
-      if (superClass != PythonType.UNKNOWN) {
-        var superClassFqn = typeFqn(moduleFqn, superClass);
+      var superClassFqn = typeFqn(moduleFqn, superClassWrapper.type());
+      if (superClassFqn != null) {
```

Comment:
I have a question that is not strictly related to this PR, but to the overall functional impact of what this piece of code is doing.

Before, we used the `hasSuperClassWithoutDescriptor` as a flag to determine whether a type hierarchy was unknown or not. (This was directly mapped to the `ClassSymbol#hasSuperClassWithoutDescriptor` field used in `ClassSymbol#hasUnresolvedTypeHierarchy`).

Now that we can resolve super class FQNs for `UnresolvedImportType`, I'm wondering how we guarantee the behavior of `ClassSymbol#hasUnresolvedTypeHierarchy` in cases like where a class inherits from another that is external to the project (and which we ultimately can't resolve).

Is there any specific measure for that?

Reply:
Added assertions on symbol model